src/notation/internal/notationinteraction.cpp

```diff
@@ -5044,8 +5080,9 @@ void NotationInteraction::addMelisma()
     mu::engraving::PlacementV placement = lyrics->placement();
     mu::engraving::PropertyFlags pFlags = lyrics->propertyFlags(mu::engraving::Pid::PLACEMENT);
     Fraction endTick = segment->tick(); // a previous melisma cannot extend beyond this point
-
+    int clipTextPos = m_clipboardTextPos;
     endEditText();
+    m_clipboardTextPos = clipTextPos;
```

Comment:
this ensures explicitly pressing _ while pasting words doesn't lose what position in the clipboard text you're up to.

Reply:
You need to do the same with - for multi-syllabic words